wellness.sql

```diff
@@ -104,18 +104,137 @@ aggregate_wellness_survey_data AS (
     GROUP BY
         site_short
 ),
---Growth KPIs and students completing Covitality KPI
-prep_all_wellness_kpis AS (
+--Average # of sessions for reb/blue Covi students or 1:1 (case notes)
+--% of students/# of sessions/amt of time that students with red and blue CoVi scores have spent receiving support/counseling/coaching for their social emotional wellbeing health (either through a workshop, small group or 1:1s)
+--Broken down into various CTEs to gather students with red/blue covi, wellness sessions attended, and case notes logged
+--Pulling students with red/blue covi from academic term (2020-21AY)
+gather_red_blue_covi_at AS (
     SELECT
-        A.site_short,
-        wellness_covi_assessment_completed_ay,
-        wellness_survey_wellness_services_assisted_num,
-        wellness_survey_wellness_services_assisted_denom
+        student_c,
+        site_short,
+        CASE
+            WHEN co_vitality_scorecard_color_c IN ('Blue', 'Red') THEN 1
+            ELSE NULL
+        END AS wellness_blue_red_denom
+    FROM
+        `data-warehouse-289815.salesforce_clean.contact_at_template`
+    WHERE
+        grade_c != '8th Grade'
+        AND college_track_status_c = '11A'
+        AND AY_NAME = "AY 2020-21"
+    GROUP BY
+        student_c,
+        site_short,
+        co_vitality_scorecard_color_c
+),
+--Sum students that have a red or blue covitality color at some point during 2020-21AY
+sum_of_blue_red_covi AS (
+    SELECT
+        site_short,
+        SUM(wellness_blue_red_denom) AS sum_of_blue_red_covi_for_avg #students with blue/red Covitality scorecard colors for denominator
     FROM
-        students_that_completed_covi AS A
-        LEFT JOIN aggregate_wellness_survey_data AS B ON A.site_short = B.site_short
-)
+        gather_red_blue_covi_at
+    GROUP BY
+        site_short
+),
+--gather Wellness sessions attended during 2020-21
+gather_wellness_attendance_data AS (
+    SELECT
+        SUM(attendance_numerator_c) AS sum_attended_wellness_sessions,
+        site_short
+    FROM
+        `data-warehouse-289815.salesforce_clean.class_template` CT
+        LEFT JOIN `data-warehouse-289815.salesforce_clean.contact_at_template` CAT ON CAT.AT_Id = CT.Academic_Semester_c
+    WHERE
+        Attendance_Numerator_c > 0
+        AND department_c = 'Wellness'
+        AND dosage_types_c NOT LIKE '%NSO%'
+        AND AY_NAME = "AY 2020-21"
+        AND grade_c != '8th Grade'
+        AND Outcome_c != 'Cancelled'
+        AND college_track_status_c = '11A'
+        AND co_vitality_scorecard_color_c IN ('Blue', 'Red')
```

Comment:
@isabelreyna isolate student groups to red/blue Covi scorecard colors in the Fall (vs. later in the year, Spring) (prev, prev prev at - e.g. Fall term look at Spring test or Look at Fall). 
Confirm which covi scorecard color to look at, and when (is Summer support offered?)- IR to connect with team lead to confirm reporting group with team lead